service/history/queues/queue_immediate.go

```diff
@@ -161,6 +168,19 @@ func (p *immediateQueue) processEventLoop() {
 	}
 }
 
+func (p *immediateQueue) processPollTimer(pollTimer *time.Timer) {
+	// NOTE: this check can actually make the maximum poll interval becomes
+	// 2 * configured maxPollInterval + jitter
+	if p.lastPollTime.Add(p.options.MaxPollInterval()).Before(p.timeSource.Now()) {
```

Comment:
due to jitter, 50% chance this will skip the process, why do we need this logic? Can we just run processNewRange() every time the pollTimer fires?

Reply:
yeah make sense. Removed this check and `lastPollTime` definition.